Vending Machine/VendingMachine/UseCases/LookUseCase.cs

```diff
@@ -0,0 +1,31 @@
+﻿using iQuest.VendingMachine.PresentationLayer;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace iQuest.VendingMachine.UseCases
+{
+    internal class LookUseCase : IUseCase
+    {
+        private readonly VendingMachineApplication application;
+        private readonly MainDisplay mainDisplay;
+        private readonly ProductRepository repository = new ProductRepository();
```

Comment:
A dependency of a class should be injected in that class, not instantiated.
The same with ShelfView. Don't make a static method in it. Inject the class where is needed.